web/wp-content/themes/bumblebee/functions.listicle.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Bumblebee functions and definitions
+ *
+ * @link https://developer.wordpress.org/themes/basics/theme-functions/
+ *
+ * @package bumblebee
+ */
+
+add_filter(
+	'the_content',
+	function( $content ) {
+		$delimiter    = '<div class="tmbi-card">';
+		$cards        = array_filter( explode( $delimiter, $content ) );
+		$total_cards  = count( $cards );
+		$current_card = 0;
+		$section_num  = 0;
+		for ( $j = 0; $j <= $total_cards - 1; $j = $j + 3 ) :
+			$section_num++;
+			?>
+			<section class="content pure-g">
+			<section class="social-share-bar-desktop pure-u-lg-2-24"></section>
+			<section class=" pure-u-1 pure-u-lg-14-24">
+				<div class="contentbar">
+					<?php
+					for ( $i = 0; $i < 3; $i++ ) :
+						if ( ( $current_card ) <= ( $total_cards - 1 ) ) :
+							get_the_card_markup( $current_card + 1, $total_cards, $cards[ $current_card + 1 ] );
+							$current_card++;
+							endif;
+						endfor;
+					?>
+					</div>
+					</section>
+					<section class="sidebar pure-u-1 pure-u-lg-8-24">
+					<?php
+					bumblebee_render_ad(
+						uniqid( 'ad' ),
+						[
+							'slot-name' => 'rail' . ( 1 === $section_num ? 'top' : 2 === $section_num ? 'middle' : 'scroll' ),
+							'sizes'     => '300x250,300x600',
+						]
+					);
+				?>
+				</section>
+			</section>
+
+			<section class="full-width-ad">
+				<?php
+				bumblebee_render_ad(
+					uniqid( 'ad' ),
+					[
+						'slot-name' => ( 1 === $section_num ? 'top' : 2 === $section_num ? 'middle' : 'scroll' ),
+						'sizes'     => '970x550,970x250,970x90,728x90,300x250,3x3',
+					]
+				);
+				?>
+			</section>
+		<?php
+		endfor;
+		return $cards;
+	},
+	11
+);
+
+/**
+ * Getting card content.
+ *
+ * @param String $current_card  current card.
+ * @param String $total_cards  total no of cards.
+ * @param String $card  card data.
+ */
+function get_the_card_markup( $current_card, $total_cards, $card ) {
+	preg_match( '|<h[^>]+>(.*)</h[^>]+>|iU', $card, $headings );
+	preg_match_all( '%(<p[^>]*>.*?</p>)%i', $card, $paragraph );
+	$image = $paragraph[0][0];
+	$dek   = $paragraph[0][1];
+	ob_start();
+	?>
+	<div class="listicle-card">
```

Comment:
Markup this complex should be in template file, so frontend devs can alter it without having to touch PHP logic.